### PR TITLE
Document Storage Queue Binding and SignalR AAD Support

### LINKED ISSUE #2613: [1.9] Document support for Azure AD auth in Azure Storage Queue and Azure SignalR bindings
**What content needs to be created or modified?**
These components are gaining support for AAD auth in 1.9:

- bindings.azure.storagequeues
- bindings.azure.signalr

Additionally, there have been a few more small changes:

- bindings.azure.storagequeues: the way custom endpoints are specified changed slightly, so it's consistent with Azure Blob Storage (does not use "production-style" URLs anymore)
- bindings.azure.signalr: added a few more (optional) metadata keys

The docs for those bindings need to be updated to reflect the new metadata keys

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**
https://github.com/dapr/components-contrib/pull/1842
https://github.com/dapr/components-contrib/pull/1852

### PR DESCRIPTION
**Pull request by berndverst**
Signed-off-by: Bernd Verst <4535280+berndverst@users.noreply.github.com>

Thank you for helping make the Dapr documentation better!

Relates to #2613 but doesn't close it. There are some more metadata options for @ItalyPaleAle to document.

### CODE REVIEW

**Comment by berndverst** [comment]
FYI @ItalyPaleAle - somehow documenting this fell through the cracks, but completed it now for you.

**Comment by ItalyPaleAle** [comment]
I was waiting for the v1.9 branch of docs to be available. Thanks for taking the lead on this :)

However, please make sure this targets v1.9 and not 1.8, as it's not yet available.

**Comment by berndverst** [comment]
Ah for some reason I thought it made it into v1.8 my bad @ItalyPaleAle

**Comment by greenie-msft** [comment]
@hhunter-ms - can you please review?

File: daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md [hunk 1]
```diff
@@ -36,9 +36,14 @@ The above example uses secrets as plain strings. It is recommended to use a secr
 
 | Field              | Required | Binding support |  Details | Example |
 |--------------------|:--------:|------------|-----|---------|
-| connectionString | Y | Output | The Azure SignalR connection string | `"Endpoint=https://<your-azure-signalr>.service.signalr.net;AccessKey=<your-access-key>;Version=1.0;"` |
+| connectionString | Y | Output | The Azure SignalR connection string | `"Endpoint=https://<your-azure-signalr>.service.signalr.net;AccessKey=<your-access-key>;Version=1.0;"`. If the connection string does not contain the `AccessKey` option Azure Active Directory authentication will be attempted. |
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md]:
```suggestion
| connectionString | Y | Output | The Azure SignalR connection string | `"Endpoint=https://<your-azure-signalr>.service.signalr.net;AccessKey=<your-access-key>;Version=1.0;"`. If the connection string does not contain the `AccessKey` option, Azure Active Directory authentication will be attempted. |
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md [hunk 2]
```diff
@@ -36,9 +36,14 @@ The above example uses secrets as plain strings. It is recommended to use a secr
...
 | hub | N | Output | Defines the hub in which the message will be send. The hub can be dynamically defined as a metadata value when publishing to an output binding (key is "hub") | `"myhub"` |
 
+### Azure Active Directory (Azure AD) authentication
+
+The Azure Cosmos DB state store component supports authentication using all Azure Active Directory mechanisms. For further information and the relevant component metadata fields to provide depending on the choice of Azure AD authentication mechanism, see the [docs for authenticating to Azure]({{< ref authenticating-azure.md >}}).
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md]:
```suggestion
The Azure Cosmos DB state store component supports authentication using all Azure Active Directory mechanisms. See the [docs for authenticating to Azure]({{< ref authenticating-azure.md >}}) to learn more about the relevant component metadata fields based on your choice of Azure AD authentication mechanism.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md [hunk 3]
```diff
@@ -36,9 +36,14 @@ The above example uses secrets as plain strings. It is recommended to use a secr
...
+
+You can read additional information for setting up Cosmos DB with Azure AD authentication in the [section below](#setting-up-cosmos-db-for-authenticating-with-azure-ad).
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md]:
```suggestion
Read more about [setting up Cosmos DB with Azure AD authentication](#setting-up-cosmos-db-for-authenticating-with-azure-ad).
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/storagequeues.md [hunk 4]
```diff
@@ -45,12 +45,18 @@ The above example uses secrets as plain strings. It is recommended to use a secr
...
 | queueEndpointUrl | N | Input/Output | Optional custom endpoint URL. This is useful, for example, with "[production-style URLs](https://github.com/Azure/azurite#production-style-url)" when using the [Storage emulator](https://github.com/Azure/azurite). This is also potentially useful when using custom domains for Azure Storage, although this hasn't been tested (yet). Other possible use-cases include integration testing, debugging, or any other scenario in which one might have the inclination to instruct this Dapr binding to direct its API requests to a specific URL. | `"https://accountName.queue.example.com:10001"` |
 
+### Azure Active Directory (Azure AD) authentication
+
+The Azure Cosmos DB state store component supports authentication using all Azure Active Directory mechanisms. For further information and the relevant component metadata fields to provide depending on the choice of Azure AD authentication mechanism, see the [docs for authenticating to Azure]({{< ref authenticating-azure.md >}}).
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/storagequeues.md]:
```suggestion
The Azure Cosmos DB state store component supports authentication using all Azure Active Directory mechanisms. See the [docs for authenticating to Azure]({{< ref authenticating-azure.md >}}) to learn more about the relevant component metadata fields based on your choice of Azure AD authentication mechanism.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/storagequeues.md [hunk 5]
```diff
@@ -45,12 +45,18 @@ The above example uses secrets as plain strings. It is recommended to use a secr
...
+
+You can read additional information for setting up Cosmos DB with Azure AD authentication in the [section below](#setting-up-cosmos-db-for-authenticating-with-azure-ad).
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/storagequeues.md]:
```suggestion
Read more about [setting up Cosmos DB with Azure AD authentication](#setting-up-cosmos-db-for-authenticating-with-azure-ad).
```

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
quick grammar pass

**Comment by berndverst** [comment]
@greenie-msft I accepted @hhunter-ms 's changes. Please note that the changes will apply to almost every single Azure component page as I had copied the language from those pages.

**Review by hhunter-ms — APPROVED** [review]

File: daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md [hunk 6]
```diff
@@ -35,9 +35,14 @@ The above example uses secrets as plain strings. It is recommended to use a secr
 
 | Field              | Required | Binding support |  Details | Example |
 |--------------------|:--------:|------------|-----|---------|
-| connectionString | Y | Output | The Azure SignalR connection string | `"Endpoint=https://<your-azure-signalr>.service.signalr.net;AccessKey=<your-access-key>;Version=1.0;"` |
+| connectionString | Y | Output | The Azure SignalR connection string | `"Endpoint=https://<your-azure-signalr>.service.signalr.net;AccessKey=<your-access-key>;Version=1.0;"`. If the connection string does not contain the `AccessKey` option, Azure Active Directory authentication will be attempted. |
 | hub | N | Output | Defines the hub in which the message will be send. The hub can be dynamically defined as a metadata value when publishing to an output binding (key is "hub") | `"myhub"` |
 
+### Azure Active Directory (Azure AD) authentication
+
+The Azure Cosmos DB state store component supports authentication using all Azure Active Directory mechanisms. See the [docs for authenticating to Azure]({{< ref authenticating-azure.md >}}) to learn more about the relevant component metadata fields based on your choice of Azure AD authentication mechanism.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md]:
```suggestion
The Azure SignalR binding component supports authentication using all Azure Active Directory mechanisms. See the [docs for authenticating to Azure]({{< ref authenticating-azure.md >}}) to learn more about the relevant component metadata fields based on your choice of Azure AD authentication mechanism.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md [hunk 7]
```diff
@@ -35,9 +35,14 @@ The above example uses secrets as plain strings. It is recommended to use a secr
...
+
+Read more about [setting up Cosmos DB with Azure AD authentication](#setting-up-cosmos-db-for-authenticating-with-azure-ad).
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md]: Not sure this link goes anywhere?

File: daprdocs/content/en/reference/components-reference/supported-bindings/storagequeues.md [hunk 8]
```diff
@@ -44,12 +44,18 @@ The above example uses secrets as plain strings. It is recommended to use a secr
...
 | queueEndpointUrl | N | Input/Output | Optional custom endpoint URL. This is useful, for example, with "[production-style URLs](https://github.com/Azure/azurite#production-style-url)" when using the [Storage emulator](https://github.com/Azure/azurite). This is also potentially useful when using custom domains for Azure Storage, although this hasn't been tested (yet). Other possible use-cases include integration testing, debugging, or any other scenario in which one might have the inclination to instruct this Dapr binding to direct its API requests to a specific URL. | `"https://accountName.queue.example.com:10001"` |
 
+### Azure Active Directory (Azure AD) authentication
+
+The Azure Cosmos DB state store component supports authentication using all Azure Active Directory mechanisms. See the [docs for authenticating to Azure]({{< ref authenticating-azure.md >}}) to learn more about the relevant component metadata fields based on your choice of Azure AD authentication mechanism.
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/storagequeues.md]:
```suggestion
The Azure Storage queue binding component supports authentication using all Azure Active Directory mechanisms. See the [docs for authenticating to Azure]({{< ref authenticating-azure.md >}}) to learn more about the relevant component metadata fields based on your choice of Azure AD authentication mechanism.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/storagequeues.md [hunk 9]
```diff
@@ -44,12 +44,18 @@ The above example uses secrets as plain strings. It is recommended to use a secr
...
+
+Read more about [setting up Cosmos DB with Azure AD authentication](#setting-up-cosmos-db-for-authenticating-with-azure-ad).
```
Comment (hhunter-ms) [inline daprdocs/content/en/reference/components-reference/supported-bindings/storagequeues.md]: Not sure if this link goes anywhere either...

**Review by hhunter-ms — CHANGES_REQUESTED** [review]
@berndverst So sorry - just a couple more changes I had missed before.

File: daprdocs/content/en/reference/components-reference/supported-bindings/storagequeues.md (hunk 9 above)
Comment (berndverst) [reply daprdocs/content/en/reference/components-reference/supported-bindings/storagequeues.md]:
```suggestion
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/storagequeues.md [hunk 10]
```diff
@@ -44,12 +44,17 @@ The above example uses secrets as plain strings. It is recommended to use a secr
...
 | queueEndpointUrl | N | Input/Output | Optional custom endpoint URL. This is useful, for example, with "[production-style URLs](https://github.com/Azure/azurite#production-style-url)" when using the [Storage emulator](https://github.com/Azure/azurite). This is also potentially useful when using custom domains for Azure Storage, although this hasn't been tested (yet). Other possible use-cases include integration testing, debugging, or any other scenario in which one might have the inclination to instruct this Dapr binding to direct its API requests to a specific URL. | `"https://accountName.queue.example.com:10001"` |
 
+### Azure Active Directory (Azure AD) authentication
+
+The Azure Storage queue binding component supports authentication using all Azure Active Directory mechanisms. See the [docs for authenticating to Azure]({{< ref authenticating-azure.md >}}) to learn more about the relevant component metadata fields based on your choice of Azure AD authentication mechanism.
+
+
```
Comment (berndverst) [inline daprdocs/content/en/reference/components-reference/supported-bindings/storagequeues.md]:
```suggestion
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md (hunk 7 above)
Comment (berndverst) [reply daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md]: Should not have copied this.

File: daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md [hunk 11]
```diff
@@ -35,9 +35,14 @@ The above example uses secrets as plain strings. It is recommended to use a secr
...
-| connectionString | Y | Output | The Azure SignalR connection string | `"Endpoint=https://<your-azure-signalr>.service.signalr.net;AccessKey=<your-access-key>;Version=1.0;"` |
+| connectionString | Y | Output | The Azure SignalR connection string | `"Endpoint=https://<your-azure-signalr>.service.signalr.net;AccessKey=<your-access-key>;Version=1.0;"`. If the connection string does not contain the `AccessKey` option, Azure Active Directory authentication will be attempted. |
 | hub | N | Output | Defines the hub in which the message will be send. The hub can be dynamically defined as a metadata value when publishing to an output binding (key is "hub") | `"myhub"` |
 
+### Azure Active Directory (Azure AD) authentication
+
+The Azure SignalR binding component supports authentication using all Azure Active Directory mechanisms. See the [docs for authenticating to Azure]({{< ref authenticating-azure.md >}}) to learn more about the relevant component metadata fields based on your choice of Azure AD authentication mechanism.
+
+Read more about [setting up Cosmos DB with Azure AD authentication](#setting-up-cosmos-db-for-authenticating-with-azure-ad).
```
Comment (berndverst) [inline daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md]:
```suggestion
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md [hunk 12]
```diff
@@ -35,9 +35,13 @@ The above example uses secrets as plain strings. It is recommended to use a secr
...
 | Field              | Required | Binding support |  Details | Example |
 |--------------------|:--------:|------------|-----|---------|
-| connectionString | Y | Output | The Azure SignalR connection string | `"Endpoint=https://<your-azure-signalr>.service.signalr.net;AccessKey=<your-access-key>;Version=1.0;"` |
+| connectionString | Y | Output | The Azure SignalR connection string | `"Endpoint=https://<your-azure-signalr>.service.signalr.net;AccessKey=<your-access-key>;Version=1.0;"`. If the connection string does not contain the `AccessKey` option, Azure Active Directory authentication will be attempted. |
 | hub | N | Output | Defines the hub in which the message will be send. The hub can be dynamically defined as a metadata value when publishing to an output binding (key is "hub") | `"myhub"` |
 
+### Azure Active Directory (Azure AD) authentication
+
+The Azure SignalR binding component supports authentication using all Azure Active Directory mechanisms. See the [docs for authenticating to Azure]({{< ref authenticating-azure.md >}}) to learn more about the relevant component metadata fields based on your choice of Azure AD authentication mechanism.
+
```
Comment (berndverst) [inline daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md:44]:
```suggestion
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md [hunk 13]
```diff
@@ -35,9 +35,12 @@ The above example uses secrets as plain strings. It is recommended to use a secr
 
 | Field              | Required | Binding support |  Details | Example |
 |--------------------|:--------:|------------|-----|---------|
-| connectionString | Y | Output | The Azure SignalR connection string | `"Endpoint=https://<your-azure-signalr>.service.signalr.net;AccessKey=<your-access-key>;Version=1.0;"` |
+| connectionString | Y | Output | The Azure SignalR connection string | `"Endpoint=https://<your-azure-signalr>.service.signalr.net;AccessKey=<your-access-key>;Version=1.0;"`. If the connection string does not contain the `AccessKey` option, Azure Active Directory authentication will be attempted. |
 | hub | N | Output | Defines the hub in which the message will be send. The hub can be dynamically defined as a metadata value when publishing to an output binding (key is "hub") | `"myhub"` |
 
+### Azure Active Directory (Azure AD) authentication
+
+The Azure SignalR binding component supports authentication using all Azure Active Directory mechanisms. See the [docs for authenticating to Azure]({{< ref authenticating-azure.md >}}) to learn more about the relevant component metadata fields based on your choice of Azure AD authentication mechanism.
```
Comment (berndverst) [inline daprdocs/content/en/reference/components-reference/supported-bindings/signalr.md:43]:
```suggestion
The Azure SignalR binding component supports authentication using all Azure Active Directory mechanisms. See the [docs for authenticating to Azure]({{< ref authenticating-azure.md >}}) to learn more about the relevant component metadata fields based on your choice of Azure AD authentication mechanism.

```

**Review by hhunter-ms — APPROVED** [review]
lgtm

**Review by greenie-msft — APPROVED** [review]

**Pull request merged**